### DISCUSSION TITLE
Font not loading on iOS mobile devices after downloading Canvas with Fabric.js

### DISCUSSION THREAD

**Discussion by parthvaghani4507** [fabricjs/fabric.js#9658] [General]
I'm encountering an issue with font loading specifically on iOS mobile devices after downloading a canvas generated using Fabric.js. The font loads perfectly on web and Android devices, but fails consistently on iOS mobile device.

[Before Download and generate canvas](https://i.stack.imgur.com/5UIBh.png)
[After Download and generate new canvas](https://i.stack.imgur.com/3SL6J.png)

**Here's a brief overview of the problem:**
**Font Types:** I've tried using both TTF and WOFF font formats.

**Preloading:** Fonts are preloaded using Fabric.js's loadFont method before rendering the canvas.

**Error Handling:** No errors are reported during font loading or canvas generation on iOS.

**Post-Download Issue:** The problem occurs only after downloading the canvas. Fonts fail to load on iOS devices even though they are correctly loaded during the initial canvas rendering.

I've thoroughly tested the application on various iOS devices and browsers, and the issue persists across all of them.

**Font load code :**

`

> templateData.text_json.forEach(textElement => {
>   var f = new FontFace(textElement.fontName, `url(${preTextfam + textElement.fontfamily})`)
>   fontLoadingPromises.push(
>     f.load().then(function (loadedFace) {
>       document.fonts.add(loadedFace);
>       return textElement;
>     }).catch(function (error) {
>       console.log(`Error loading font ${textElement.fontName}:`, error);
>       // Use a fallback font if custom font fails to load
>       textElement.fontName = "Inter-Regular"; // Fallback font
>       return textElement;
>     })
>   );
> });

`
**Download code**

`
> function downloadImage(format, filename) {
>   const dataURL = canvas.toDataURL({
>     format: format,
>     multiplier: 1,
>     quality: 1, // Adjust as needed for JPG
>   });
>   var a = document.createElement("a");
>   a.href = dataURL;
>   a.download = "download." + format;
>   document.body.appendChild(a);
>   a.click();
>   document.body.removeChild(a);
>   canvas.overlayImage?.dispose();
>   canvas.setOverlayImage('', canvas.renderAll.bind(canvas));
> }
> // Event handler for downloading PNG
> $("#download_png").on("click", function () {
>   addWatermarkAndDownload('png', 'download.png');
> });

`

**Comment by asturur** [comment]
In your code is not clear if your are  loading the fonts before the Text instance is created or after.
But anyway since as you mentioned your code works fine everywhere but not on IOS i would argue is a IOS issue not a fabricJS one

**Reply by parthvaghani4507** [reply]
![Capture](https://github.com/fabricjs/fabric.js/assets/68509180/7af61850-72a0-4f29-8f60-a94ea49d32f9)
Please check this

**Reply by asturur** [reply]
Yes what yo do seems correct.
